astropy/io/ascii/core.py

```diff
@@ -1088,17 +1088,21 @@ def _convert_vals(self, cols):
                 try:
                     col.data = converter_func(col.str_vals)
                     col.type = converter_type
-                except (TypeError, ValueError) as err:
-                    col.converters.pop(0)
-                    last_err = err
-                except OverflowError as err:
+                except (OverflowError, TypeError, ValueError) as err:
                     # Overflow during conversion (most likely an int that
                     # doesn't fit in native C long). Put string at the top of
                     # the converters list for the next while iteration.
-                    warnings.warn(
-                        "OverflowError converting to {} in column {}, reverting to String."
-                        .format(converter_type.__name__, col.name), AstropyWarning)
-                    col.converters.insert(0, convert_numpy(numpy.str))
+                    # With python/cpython#95778 this has been replaced by a
+                    # ValueError: Exceeds the limit (4300) for integer string conversion
+                    # so need to catch that as well.
+                    if type(err) == OverflowError or (type(err) == ValueError and
```

Comment:
Why not use isinstance?

Reply:
Because I was just blindly copying from the Python docs? 😆
But test_read is passing in 3.11; have to see if any of the other failures are popping up here.

Reply:
I assume isinstance might be faster?

Reply:
I don't know which is faster. I am just wondering about the choice of one style over another.